src/libraries/System.Private.Xml/src/System/Xml/Core/XmlSubtreeReader.cs

```diff
@@ -66,7 +73,7 @@ private enum State
 
         // namespace management
         private readonly XmlNamespaceManager _nsManager;
-        private NodeData[] _nsAttributes;
+        private NodeData?[]? _nsAttributes;
```

Comment:
yes and no: for all valid indices the entries are not null but for any other indices they may be null. I'll change it though since that seems much cleaner - I feel like the code is polluted with `!` for the sake of being correct and combined with https://github.com/dotnet/roslyn/issues/34644 that makes it more annoying than helpful.